File: SDKs/parliament.uk/members/v1/python/src/sdk/models/shared/registeredinterestcategorylistitem.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from . import link
from . import registeredinterestcategory


@dataclass_json
@dataclass
class RegisteredInterestCategoryListItem:
    links: Optional[List[link.Link]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'links' }})
    value: Optional[List[registeredinterestcategory.RegisteredInterestCategory]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'value' }})
    

File: SDKs/parliament.uk/members/v1/typescript/src/sdk/models/shared/registeredinterestcategorylistitem.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { Link } from "./link";
import { RegisteredInterestCategory } from "./registeredinterestcategory";



export class RegisteredInterestCategoryListItem extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=links", elemType: Link })
  links?: Link[];

  @SpeakeasyMetadata({ data: "json, name=value", elemType: RegisteredInterestCategory })
  value?: RegisteredInterestCategory[];
}
